### PR TITLE
[risk=low][no ticket] Upgrade many UI packages

### PR DESCRIPTION
**Pull request by jmthibault79**
Update all babel packages to latest
Upgrade eslint (and related) to latest 
Upgrade prettier to latest 
Upgrade typescript to latest
Upgrade fontawesome to latest

Upgrade all jest packages to the latest 27.x

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "@fortawesome/react-fontawesome": "^0.1.16",
     "@types/color": "^3.0.2",
     "@types/enzyme": "^3.10.10",
-    "@types/jest": "^26.0.21",
```
Comment (jmthibault79) [inline ui/package.json:35]: There is a version 28 available but we depend on many packages which use v27, and our code is not yet compatible v28 anyway

File: ui/package.json [hunk 2]
```diff
@@ -32,7 +32,6 @@
...
```
Comment (jmthibault79) [inline ui/package.json:35]: does not appear to be needed

**Comment by jmthibault79** [comment]
1 e2e failure - notebook copy timeout for duplicated workspace

**Review by evrii — APPROVED** [review]
Looks good to me.
I ran locally without a problem.
I looked over the breaking changes in jest 27, but nothing looks too concerning.
https://github.com/facebook/jest/blob/main/CHANGELOG.md#2700
The font awesome changes also seem like they should not be a problem.

**Pull request merged**